packages/sd-jwt/src/action-handler.ts

```diff
@@ -0,0 +1,227 @@
+      return context.agent.keyManagerSign({ keyRef: key.kid, data })
+    }
+
+    const sdjwt = new SDJwtVcInstance({
+      hasher: this.algorithms.hasher,
+      saltGenerator: this.algorithms.saltGenerator,
+      kbSigner: signer,
+      kbSignAlg: alg,
+    })
+    const credential = await sdjwt.present(
+      args.presentation,
+      args.presentationFrame as PresentationFrame<SdJwtVcPayload>,
+      { kb: args.kb },
+    )
+    return { presentation: credential }
+  }
+
+  /**
+   * Verify a signed SD-JWT credential.
+   * @param args - Arguments necessary for the verify a SD-JWT credential.
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   * @returns
+   */
+  async verifySdJwtVc(args: IVerifySdJwtVcArgs, context: IRequiredContext): Promise<IVerifySdJwtVcResult> {
+    // biome-ignore lint/style/useConst: <explanation>
+    let sdjwt: SDJwtVcInstance
+    const verifier: Verifier = async (data: string, signature: string) =>
+      this.verify(sdjwt, context, data, signature)
+
+    sdjwt = new SDJwtVcInstance({ verifier, hasher: this.algorithms.hasher })
+    const verifiedPayloads = await sdjwt.verify(args.credential)
+
+    return { verifiedPayloads }
+  }
+
+  /**
+   * Verify the key binding of a SD-JWT by validating the signature of the key bound to the SD-JWT
+   * @param sdjwt - SD-JWT instance
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   * @param data - signed data
+   * @param signature - The signature
+   * @param payload - The payload of the SD-JWT
+   * @returns
+   */
+  private verifyKb(sdjwt: SDJwtVcInstance, context: IRequiredContext, data: string, signature: string, payload: JwtPayload): Promise<boolean> {
+    if (!payload.cnf) {
+      throw Error('other method than cnf is not supported yet')
+    }
+    const key = payload.cnf.jwk as JsonWebKey
+    return this.algorithms.verifySignature(data, signature, key)
+  }
+
+  /**
+   * Validates the signature of a SD-JWT
+   * @param sdjwt - SD-JWT instance
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   * @param data - signed data
+   * @param signature - The signature
+   * @returns
+   */
+  async verify(sdjwt: SDJwtVcInstance, context: IRequiredContext, data: string, signature: string) {
+    const decodedVC = await sdjwt.decode(`${data}.${signature}`)
+    const issuer: string = ((decodedVC.jwt as Jwt).payload as Record<string, unknown>).iss as string
+    if (!issuer.startsWith('did:')) {
+      throw new Error('invalid_issuer: issuer must be a did')
+    }
+    const didDoc = await context.agent.resolveDid({ didUrl: issuer })
+    if (!didDoc) {
+      throw new Error('invalid_issuer: issuer did not resolve to a did document')
+    }
+    const didDocumentKey = didDoc.didDocument?.verificationMethod?.find((key) => key.id)
```

Comment:
This basically does nothing except for checking there is a verificationMethod with an id. That is way too naive. It should be checking for an assertionMethod. Suggest to create a TODO we need to fix this

Reply:
added a TODO